File: frontend/types/github.ts
export interface UserProfile {
  login: string;
  name: string | null;
  avatar_url: string;
  bio: string | null;
  public_repos: number;
  followers: number;
  following: number;
  created_at: string;
}

export interface LanguageStat {
  name: string;
  bytes: number;
  percentage: number;
}

export interface CommitActivity {
  week: number;
  days: number[];
  total: number;
}

export interface PRVelocity {
  number: number;
  title: string;
  state: string;
  created_at: string;
  closed_at: string | null;
  merged_at: string | null;
  turnaround_hours: number | null;
}

export interface DevRhythm {
  hour: number;
  commit_count: number;
}

export interface AnalyticsSummary {
  profile: UserProfile;
  languages: LanguageStat[];
  commit_activity: CommitActivity[];
  pr_velocity: PRVelocity[];
  dev_rhythm: DevRhythm[];
  avg_pr_turnaround_hours: number | null;
  total_commits_last_year: number;
  current_streak: number;
  longest_streak: number;
}

File: backend/app/schemas/github.py
from pydantic import BaseModel
from typing import Optional


class UserProfile(BaseModel):
    login: str
    name: Optional[str]
    avatar_url: str
    bio: Optional[str]
    public_repos: int
    followers: int
    following: int
    created_at: str


class LanguageStat(BaseModel):
    name: str
    bytes: int
    percentage: float


class CommitActivity(BaseModel):
    week: int
    days: list[int]
    total: int


class PRVelocity(BaseModel):
    number: int
    title: str
    state: str
    created_at: str
    closed_at: Optional[str]
    merged_at: Optional[str]
    turnaround_hours: Optional[float]


class DevRhythm(BaseModel):
    hour: int
    commit_count: int


class AnalyticsSummary(BaseModel):
    profile: UserProfile
    languages: list[LanguageStat]
    commit_activity: list[CommitActivity]
    pr_velocity: list[PRVelocity]
    dev_rhythm: list[DevRhythm]
    avg_pr_turnaround_hours: Optional[float]
    total_commits_last_year: int
    current_streak: int
    longest_streak: int
